### PR TITLE
Detects coupler when propagating contingency in sensitivity analysis

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?**
Yes, first step to solve issue #289. This doesn't fix that issue, but detects the problematic contingency, gives a warning and does not add it to the propagated contingencies to avoid giving wrong results.

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java [hunk 1]
```diff
@@ -46,55 +49,88 @@ public int getIndex() {
...
+                    throw new UnsupportedOperationException("TODO");
             }
-
         }
-        return propagatedContingencies;
+
+        return new PropagatedContingency(contingency, index, branchIdsToOpen, hvdcIdsToOpen, switchesToOpen, terminalsToDisconnect);
+    }
+
+    private static boolean isCoupler(Switch s) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:130]: `hasBusBarSectionCoulper` could be better here, what do you think?

File: src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java [hunk 2]
```diff
@@ -46,55 +49,88 @@ public int getIndex() {
...
-                        //TODO: support all kinds of contingencies
-                        throw new UnsupportedOperationException("TODO");
-                }
+            PropagatedContingency propagatedContingency = PropagatedContingency.create(network, contingency, index);
+            Optional<Switch> coupler = propagatedContingency.switchesToOpen.stream().filter(PropagatedContingency::isCoupler).findFirst();
+            if (coupler.isEmpty()) {
+                propagatedContingencies.add(propagatedContingency);
+            } else {
+                // Sensitivity analysis works in bus view, it cannot deal (yet)  with contingencies whose propagation encounters a coupler
+                LOGGER.warn("Propagated contingency '{}' not processed: coupler '{}' has been encountered while propagating the contingency",
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:81]: I don't know if coupler has a meaning in English indeed. But for sure you have to precise that it is between two bus bar sections.

File: src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:130]: for me coupler already means busbar section coupler

File: src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:81]: Coupler is a well known term in power system

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:81]: Here you skip the contingency, so it means that the result will not contain this contingency. The other solution is to set a predefined result. What is preferable?

**Pull request merged**